pages/activity/[id].js

```diff
@@ -1,12 +1,42 @@
 import { useRouter } from "next/router";
 import ActivityDetails from "@/components/layout/ActivityDetails";
+import Button from "@/components/ui/Button";
+import { useState } from "react";
+import ActivityForm from "@/components/layout/ActivityForm";
+import styled from "styled-components";
 
-export default function ActivityPage({ activities }) {
+export default function ActivityPage({ activities, handleActivity }) {
   const router = useRouter();
   const { id } = router.query;
   const activity = activities.find((activity) => activity.id === id);
+  const [showForm, setShowForm] = useState(false);
 
   if (!activity) return <p>Loading...</p>;
+  function handleToggleEdit() {
+    setShowForm(!showForm);
+  }
 
-  return <ActivityDetails {...activity} />;
+  return (
+    <>
+      {!showForm ? (
+        <StyledSection>
+          <Button onClick={() => setShowForm(true)} isPrimary>
+            Edit activity
+          </Button>
+        </StyledSection>
+      ) : (
+        <ActivityForm
+          setShowForm={setShowForm}
```

Comment:
Again instead of passing the setter function better pass the handleToggleEdit(). Or to be more precise create a `function closeForm(){setShowForm(false)}`.
I think handleToggleEdit() is not used at the moment.